ci/ray_ci/test_linux_tester_container.py

```diff
@@ -37,7 +38,8 @@ def test_enough_gpus() -> None:
         assert False, "Should not raise an AssertionError"
 
 
-def test_run_tests_in_docker() -> None:
+@mock.patch.object(tempfile, "mkdtemp")
```

Comment:
why do you need to mock `mkdtemp`?

could you just inject the `get_artifact_mount()` in the test setup? then it can create a temp dir for real?

generally speaking, mocking an external rpc call (e.g. s3 api call) is okay, but we prefer to run filesystem calls for real in tests.